sdccc/src/test/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditionsTest.java

```diff
@@ -435,6 +445,247 @@ private void setupMetricStatusManipulation(final MetricCategory metricCategory,
+                .thenReturn(Optional.of(mockAlertConditionState2));
+        when(mockMdibAccess.getState(ALERT_CONDITION_HANDLE, AlertConditionState.class))
+                .thenReturn(Optional.of(mockAlertConditionState));
+        when(mockMdibAccess.getState(ALERT_CONDITION_HANDLE2, AlertConditionState.class))
+                .thenReturn(Optional.of(mockAlertConditionState2));
+
+        when(mockAlertSystemState.getActivationState()).thenReturn(AlertActivation.OFF);
+        when(mockAlertSystemState2.getActivationState()).thenReturn(AlertActivation.OFF);
+        when(mockAlertConditionState.getActivationState()).thenReturn(AlertActivation.OFF);
+        when(mockAlertConditionState2.getActivationState()).thenReturn(AlertActivation.OFF);
+
+        when(mockAlertConditionState.isPresence()).thenReturn(true);
+        when(mockAlertConditionState2.isPresence()).thenReturn(true);
+    }
+
+    private void verifyAlertConditionPresenceSetAlertActivationInteractions(
+            final int numberOfManipulations,
+            final List<String> expectedHandles,
+            final List<AlertActivation> expectedAlertActivations) {
+        final var activationStateHandleCaptor = ArgumentCaptor.forClass(String.class);
+        final var activationStateCaptor = ArgumentCaptor.forClass(AlertActivation.class);
+        verify(mockManipulations, times(numberOfManipulations))
+                .setAlertActivation(activationStateHandleCaptor.capture(), activationStateCaptor.capture());
+
+        assertEquals(expectedHandles, activationStateHandleCaptor.getAllValues());
+        assertEquals(expectedAlertActivations, activationStateCaptor.getAllValues());
+    }
+
+    private void verifyAlertConditionPresenceSetAlertConditionPresenceInteractions(
+            final int numberOfManipulations,
+            final List<String> expectedHandles,
+            final List<Boolean> expectedAlertActivations) {
+        final var handleCaptor = ArgumentCaptor.forClass(String.class);
+        final var presenceCaptor = ArgumentCaptor.forClass(Boolean.class);
+        verify(mockManipulations, times(numberOfManipulations))
+                .setAlertConditionPresence(handleCaptor.capture(), presenceCaptor.capture());
+
+        assertEquals(expectedHandles, handleCaptor.getAllValues());
+        assertEquals(expectedAlertActivations, presenceCaptor.getAllValues());
+    }
+
+    @Test
+    @DisplayName(
+            "AlertConditionPresence: set alert activation and set presence correctly and stop when a alert condition was successfully manipulated")
+    void testSetPresenceForAlertConditionSuccessful() {
+        alertConditionPresenceManipulationSetup();
+        when(mockManipulations.setAlertActivation(anyString(), eq(AlertActivation.OFF)))
+                .thenReturn(ResponseTypes.Result.RESULT_SUCCESS);
+        when(mockManipulations.setAlertConditionPresence(anyString(), eq(true)))
+                .thenReturn(ResponseTypes.Result.RESULT_SUCCESS)
```

Comment:
the second thenReturn() is not neccessary.

Reply:
Removed it